main.tf

```diff
@@ -433,7 +433,10 @@ resource "aws_efs_file_system" "this" {
 
 resource "aws_efs_mount_target" "this" {
   # we coalescelist in order to specify the resource keys when we create the subnets using the VPC or they're specified for us.  This works around the for_each value depends on attributes which can't be determined until apply error
-  for_each = zipmap(coalescelist(var.private_subnets, var.private_subnet_ids), local.private_subnet_ids)
+  for_each = {
+    for k, v in zipmap(coalescelist(var.private_subnets, var.private_subnet_ids), local.private_subnet_ids) : k => v
+    if var.enable_ephemeral_storage == false
```

Comment:
I created an example which `github-ephemeral-storage` which uses ephemeral storage and works with the code as written.

Reply:
> 

shouldn't this be `if ! var.enable_ephemeral_storage`? We only want to make this map if we need to mount EFS.

Reply:
oh thats my bad, didn't understand the variable fully. this is not great - we should have never accepted the `enable_ephemeral_storage` and instead it should have been `enable_efs_storage` which is false by default and users then opt in. reading through the module now, everything is backwards and confusing

Reply:
I struggled with how to simultaneously support ephemeral storage, EFS, and whatever Amazon calls the capacity constrained storage that Fargate provides by default.

Ultimately, I decided to  try to encourage EFS adoption because it fixes #206.  In doing that, I unintentionally broke ephemeral storage users (who apparently aren't too worried about #206 ).

So... Where to go from here?

I agree with @llamahunter.  Merge #261 so we can fix stuff for the people who rely on the ephemeral storage and #257 so we're safer by default.

That doesn't resolve the everything is "backwards and confusing problem" (which is a pretty good way to describe it :)

Long term, I'd like to see us fix #19.  Doing so would allow us to remove a lot of complexity (and cost) from this module since we wouldn't need ALB or ephemeral storage support.

Reply:
Not sure I understand the whole picture, but I have executed `examples/github-complete` with `enable_ephemeral_storage` set as `true` and `false`. Both cases created some resources.

@bryantbiggs Should this be merged as it is, or do we still need something implemented in this PR?